src/util/address.rs

```diff
@@ -522,6 +523,50 @@ impl Address {
         }
     }
 
+    /// Create a pay to taproot address from an untweaked key
+    /// 
+    /// This is done by using the equation Q = P + H(P|c)G, where 
+    ///  * Q is the tweaked key
+    ///  * P is the internal key
+    ///  * H is the hash function
+    ///  * c is the commitment data
+    ///  * G is the generator point
+    pub fn dangerous_p2tr<C: secp256k1::Verification>(
```

Comment:
I think this should have bold warning that the function is dangerous and an explanation why.

Reply:
This is not the dangerous API. The current existing API is the dangerous one. Operating on an untweaked key and tweaking it is okay. 

But the current API operates on an untweaked key, does not tweak it, and still gives an address. The current existing API should be renamed to dangerous_p2tr and we need a new p2tr that works on untweaked keys.